lib/redcord/relation.rb

```diff
@@ -137,6 +143,30 @@ def count
 
   private
 
+  sig { returns(T.nilable(String)) }
+  def extract_hash_tag!
+    attr = model.class_variable_get(:@@shard_by_attribute)
+    return nil if attr.nil?
+
+    if !query_conditions.keys.include?(attr)
+      raise "Queries must contain attribute '#{attr}' since model #{model.name} is sharded by this attribute"
+    end
+
+    condition = query_conditions[attr]
+    case condition
+    when Integer, String
+      "{#{condition}}"
+    when Array
+      if condition.size != 2 || condition.first != condition.last
```

Comment:
I see that you forced the shard by attribute to be a normal index attribute, is this still needed?